src/services/download-service.ts

```diff
@@ -12,123 +13,153 @@ class DownloadService {
+    if (!imageRequestUrl) {
+      return await this.getImageDownloadMetadata();
     }
-
-    const requestUrl = this.getRequestUrl(config);
-    if (!requestUrl) {
-      return null;
+    if (!imageRequestUrl.includes('/ks/')) {
+      const ks = this.player.config.session?.ks;
+      imageRequestUrl = ks ? `${imageRequestUrl}/ks/${ks}` : imageRequestUrl;
     }
-
     try {
-      const response = await fetch(requestUrl, {method: 'HEAD'});
-      if (!response.ok) {
-        return null;
-      }
+      const response = await fetch(imageRequestUrl, {method: 'HEAD'});
+      return response.url;
+    } catch (e: any) {
+      this.logger.warn('Failed to get image file from url: ', imageRequestUrl);
+      // in case HEAD request failed for raw service, retry to get the image download metadata using thumbnail service
+      return await this.getImageDownloadMetadata();
+    }
+  }
 
-      const downloadUrl = response.url;
+  private async getKalturaAssets(): Promise<object> {
+    const entryId = this.player.config.sources.id;
+    const ks = this.player.config.session?.ks || '';
 
-      const isContentTypeSupported = this.player.isImage() || this.isContentTypeSupported(response);
-      const fileName = this.getFilename(response);
+    const data: Map<string, any> = await this.player.provider.doRequest(
+      [
+        {loader: CaptionsLoader, params: {entryId}},
+        {loader: FlavorsLoader, params: {entryId}},
+        {loader: AttachmentsLoader, params: {entryId}}
+      ],
+      ks
+    );
 
-      if (isContentTypeSupported && fileName) {
-        return {
-          downloadUrl,
-          fileName
-        };
+    return this.parseDataFromResponse(data);
+  }
+
+  private parseDataFromResponse(data: Map<string, any>): object {
+    const kalturaAssets = {
+      captions: [],
+      flavors: [],
+      attachments: []
+    };
+
+    if (data) {
+      if (data.has(CaptionsLoader.id)) {
+        const captionsLoader = data.get(CaptionsLoader.id);
+        kalturaAssets.captions = captionsLoader?.response?.captions;
       }
-    } catch (e: any) {
-      this.logger.warn('Failed to get file from url: ', requestUrl);
-      // in case HEAD request failed for raw service (image use-case), retry to get the image download metadata using thumbnail service
-      if (this.player.isImage()) {
-        return await this.getImageDownloadMetadata();
+      if (data.has(FlavorsLoader.id)) {
+        const flavorsLoader = data.get(FlavorsLoader.id);
+        kalturaAssets.flavors = flavorsLoader?.response?.flavors;
+      }
+      if (data.has(AttachmentsLoader.id)) {
+        const attachmentsLoader = data.get(AttachmentsLoader.id);
+        kalturaAssets.attachments = attachmentsLoader?.response?.attachments;
       }
     }
 
-    return null;
+    return kalturaAssets;
   }
 
-  private async getImageDownloadMetadata(): Promise<DownloadMetadata> {
+  private async getImageDownloadMetadata(): Promise<string> {
```

Comment:
why is this async ?

Reply:
because it is doing fetch and i need to wait for the response